civihr_employee_portal/js/reports.js

```diff
@@ -795,46 +812,90 @@
+     *   all behaviours.
+     */
     attach: function (context, settings) {
-      var that = this;
       this.instance = new HRReport();
       this.instance.initAngular();
+      this.bindReportConfigurationEvents();
+      this.bindTabsEvents();
+      this.switchToTabSpecifiedOnTheUrl();
+      this.displayConfigurationOptionsIfConfigurationsHaveBeenSaved();
+    },
+    /**
+     * Binds report configuration events for changing, saving, updating and
+     * deleting them.
+     */
+    bindReportConfigurationEvents: function () {
+      var self = this;
 
-      // Tabs bindings
-      $('.report-tabs a').bind('click', function (e) {
-        $('.report-tabs li').removeClass('active');
-        $(this).parent().addClass('active');
-        $('.report-block').addClass('hidden');
-        $('.report-block.' + $(this).data('tab')).removeClass('hidden');
-      });
-
-      switchTabsOnLoad();
-
-      // Reports configuration bindings
       $('.report-config-select').bind('change', function (e) {
-        that.instance.configGet();
+        self.instance.configGet();
       });
       $('.report-config-save-btn').bind('click', function (e) {
-        that.instance.configSave();
+        self.instance.configSave();
       });
       $('.report-config-save-new-btn').bind('click', function (e) {
-        that.instance.configSaveNew();
+        self.instance.configSaveNew(function () {
+          self.displayConfigurationOptionsIfConfigurationsHaveBeenSaved();
+        });
       });
       $('.report-config-delete-btn').bind('click', function (e) {
-        that.instance.configDelete();
+        self.instance.configDelete(function () {
+          self.displayConfigurationOptionsIfConfigurationsHaveBeenSaved();
+        });
       });
-
-      /**
-       * Switch to correct tab, on page load
-       */
-      function switchTabsOnLoad () {
-        var tabSelector = '.report-tabs a';
-        var hash = window.location.hash;
-
-        hash ? tabSelector += '[data-tab="' + hash.substr(1) + '"]' : tabSelector += ':first';
-
-        $(tabSelector).click();
+    },
+    /**
+     * Bind tab switching events for the report tabs.
+     */
+    bindTabsEvents: function () {
+      $('.report-tabs a').bind('click', function (e) {
+        $('.report-tabs li').removeClass('active');
+        $(this).parent().addClass('active');
+        $('.report-block').addClass('hidden');
+        $('.report-block.' + $(this).data('tab')).removeClass('hidden');
+      });
+    },
+    /**
+     * Displays the Report Configuration's save/update/delete options depending
+     * on the availability of configurations. If no configurations are available
+     * only the save option is available and the others are hidden.
+     */
+    displayConfigurationOptionsIfConfigurationsHaveBeenSaved: function () {
+      var deleteOption, hasSavedConfigurations, updateOption;
+
+      deleteOption = $('.report-config-delete-btn');
```

Comment:
```
var deleteOption = $('.report-config-delete-btn');
var updateOption = $('.report-config-save-btn');
var hasSavedConfigurations = $('.report-config-select option').length >= 2;
```
This way you dont need to instantiate undefined vars and assign them in the next line. Also this will save you 2 lines.